Fix:

conftest.c:28:7: error: implicitly declaring library function 'exit' with type 'void (int) __attribute__((noreturn))' [-Werror,-Wimplicit-function-declaration]

dcfldd.c:687:9: error: implicit declaration of function 'init_hashlist' is invalid in C99 [-Werror,-Wimplicit-function-declaration]

translate.c:160:10: error: implicit declaration of function 'bit_count' is invalid in C99 [-Werror,-Wimplicit-function-declaration]

util.c:85:29: error: implicit declaration of function 'safe_read' is invalid in C99 [-Werror,-Wimplicit-function-declaration]

--- configure.orig	2006-12-19 10:07:01.000000000 -0600
+++ configure	2021-02-13 23:32:31.000000000 -0600
@@ -3527,8 +3527,8 @@
   for (i = 0; i < 256; i++)
     if (XOR (islower (i), ISLOWER (i))
 	|| toupper (i) != TOUPPER (i))
-      exit(2);
-  exit (0);
+      return 2;
+  return 0;
 }
 _ACEOF
 rm -f conftest$ac_exeext
--- hash.h.orig	2005-06-26 18:17:46.000000000 -0500
+++ hash.h	2021-02-13 23:48:43.000000000 -0600
@@ -98,6 +98,7 @@
 extern void hash_update(hashlist_t *, void *, size_t);
 extern void hash_update_buf(hashlist_t *, int, int, void *, size_t);
 extern void hash_remainder(hashlist_t *, int);
+extern void init_hashlist(hashlist_t **hashlist, hashflag_t flags);
 
 /* inner hashl_* funcitons are for iterating over hashlists */
 extern void hashl_init(hashlist_t *, int);
--- translate.c.orig	2005-05-15 15:15:28.000000000 -0500
+++ translate.c	2021-02-13 23:53:59.000000000 -0600
@@ -23,6 +23,7 @@
 
 #include "dcfldd.h"
 #include "log.h"
+#include "util.h"
 
 /* Output representation of newline and space characters.
    They change if we're converting to EBCDIC.  */
--- util.c.orig	2006-02-12 08:45:16.000000000 -0600
+++ util.c	2021-02-13 23:57:33.000000000 -0600
@@ -40,6 +40,7 @@
 #include "config.h"
 #include <unistd.h>
 #include <errno.h>
+#include "safe-read.h"
 
 int buggy_lseek_support(int fdesc)
 {
